src/core/lib/backoff/backoff.cc

```diff
@@ -53,17 +56,22 @@ static double generate_uniform_random_number_between(uint32_t *rng_state,
   return a + generate_uniform_random_number(rng_state) * range;
 }
 
-grpc_millis grpc_backoff_step(grpc_exec_ctx *exec_ctx, grpc_backoff *backoff) {
+grpc_backoff_deadlines grpc_backoff_step(grpc_exec_ctx *exec_ctx,
+                                         grpc_backoff *backoff) {
   backoff->current_backoff = (grpc_millis)(GPR_MIN(
       backoff->current_backoff * backoff->multiplier, backoff->max_backoff));
+
   const double jitter = generate_uniform_random_number_between(
       &backoff->rng_state, -backoff->jitter * backoff->current_backoff,
       backoff->jitter * backoff->current_backoff);
-  const grpc_millis current_deadline =
-      grpc_exec_ctx_now(exec_ctx) +
-      (grpc_millis)(backoff->current_backoff + jitter);
-  return GPR_MAX(current_deadline,
-                 grpc_exec_ctx_now(exec_ctx) + backoff->min_backoff);
+
+  const grpc_millis current_timeout =
+      GPR_MAX((grpc_millis)(backoff->current_backoff + jitter),
+              backoff->min_connect_timeout);
+
+  const grpc_millis now = grpc_exec_ctx_now(exec_ctx);
+  return (grpc_backoff_deadlines){now + current_timeout,
+                                  now + backoff->current_backoff};
```

Comment:
now + backoff->current_backoff + jitter

Reply:
Good catch. It makes me uneasy that, while it makes sense to add the jitter to the `next_backoff_deadline` field, that's nowhere in the spec. Still, it makes sense that the deadline of a potential re-attempt be shifted by the jitter as well.